File: app/src/main/java/top/maweihao/weather/entity/BaiDu/BaiDuChoosePositionBean.java
package top.maweihao.weather.entity.BaiDu;

/**
 * 根据城市位知名，百度定位返回的json bean
 * 根据json数据，自动生成
 * Created by limuyang on 2017/6/10.
 */

public class BaiDuChoosePositionBean {

    /**
     * status : 0
     * result : {"title":{"lng":103.91332638625914,"lat":25.3600574715727},"precise":0,"confidence":14,"level":"区县"}
     */

    private int status;
    private ResultBean result;

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public ResultBean getResult() {
        return result;
    }

    public void setResult(ResultBean result) {
        this.result = result;
    }

    public static class ResultBean {
        /**
         * title : {"lng":103.91332638625914,"lat":25.3600574715727}
         * precise : 0
         * confidence : 14
         * level : 区县
         */

        private LocationBean location;
        private int precise;
        private int confidence;
        private String level;

        public LocationBean getLocation() {
            return location;
        }

        public void setLocation(LocationBean location) {
            this.location = location;
        }

        public int getPrecise() {
            return precise;
        }

        public void setPrecise(int precise) {
            this.precise = precise;
        }

        public int getConfidence() {
            return confidence;
        }

        public void setConfidence(int confidence) {
            this.confidence = confidence;
        }

        public String getLevel() {
            return level;
        }

        public void setLevel(String level) {
            this.level = level;
        }

        public static class LocationBean {
            /**
             * lng : 103.91332638625914
             * lat : 25.3600574715727
             */

            private double lng;
            private double lat;

            public double getLng() {
                return lng;
            }

            public void setLng(double lng) {
                this.lng = lng;
            }

            public double getLat() {
                return lat;
            }

            public void setLat(double lat) {
                this.lat = lat;
            }
        }
    }
}

File: app/src/main/java/top/maweihao/weather/entity/BaiDu/BaiDuChoosePositionBean.kt
package top.maweihao.weather.entity.BaiDu
import com.google.gson.annotations.SerializedName

/**
 * 根据城市位知名，百度定位返回的json bean
 * 根据json数据，自动生成
 * Created by limuyang on 2017/6/10.
 * example: http://api.map.baidu.com/geocoder/v2/?address=%E5%8D%97%E4%BA%AC&output=json&ak=RUUaavVwpwqYdz8QGHLXSFBfwhs2ba6j&mcode=B6:F3:EA:31:1E:7F:BB:C5:D2:6C:5A:A7:CA:D8:4F:DF:79:46:C9:C2;top.maweihao.weather
 */

data class BaiDuChoosePositionBean(
    @SerializedName("status") val status: Int = 0, //0
    @SerializedName("result") val result: Result = Result()
){
    data class Result(
            @SerializedName("location") val location: Location = Location(),
            @SerializedName("precise") val precise: Int = 0, //0
            @SerializedName("confidence") val confidence: Int = 0, //12
            @SerializedName("level") val level: String = "" //城市
    ){
        data class Location(
                @SerializedName("lng") val lng: Double = 0.0, //118.77807440802562
                @SerializedName("lat") val lat: Double = 0.0 //32.05723550180587
        )
    }
}



